datadotworld/files.py

```diff
@@ -99,22 +104,74 @@ def write(self, value):
                 self._queue.put(value.encode('utf-8'))
             else:
                 raise TypeError(
-                    "write() argument must be str, not {}"
-                        .format(type(value)))
+                    "write() argument must be str, not {}".format(
+                        type(value)))
         elif 'wb' == self._mode:
-            if (isinstance(value, (bytes, bytearray))):
+            if isinstance(value, (bytes, bytearray)):
                 self._queue.put(value)
             else:
                 raise TypeError(
-                    "a bytes-like object is required, not {}"
-                        .format(type(value)))
+                    "a bytes-like object is required, not {}".format(
+                        type(value)))
         else:
             raise NotImplementedError(
-                "modes other than 'w' and 'wb' not supported")
+                "write only available for files opened in write mode")
+
+    def read(self):
+        """
+        read the contents of the file that's been opened in read mode
+        """
+        if self._mode.find('r') >= 0:
+            return self._read_response.text
+        else:
+            raise NotImplementedError(
+                "read only available for files opened in read mode")
+
+    def __iter__(self):
+        """
+        the file object must be able to act as an iterable of lines to be
+        used similarly to how a local file would be used e.g. in CSV
+        reading
+        """
+        return (line.decode('utf-8') for line in
```

Comment:
I see a few potential issues here:
1) AFAIK, encoding is defined in the uploaded file, by the user, and is not guaranteed to be utf-8 coming out of our platform. We may need to detect it here (https://stackoverflow.com/questions/436220/determine-the-encoding-of-text-in-python).
2) This seems to assume files are always text and can be read line-by-line, which will fail if the user attempts to read a binary file. Perhaps we should support binary mode and use request's `iter_content()` instead of `iter_lines()` for the binary case? If preferred, we should more explicitly indicate that `rb` is not supported, by raising a `NotImplementedError` in `__init__`